dissector/v2g.lua

```diff
@@ -12,212 +12,245 @@ local V2GTP_HEADER_LENGTH = 8
 }
 
 local V2GTP_EXIDISSECTOR_DEFAULT = 0
 local V2GTP_EXIDISSECTOR_DIN = 1
 local V2GTP_EXIDISSECTOR_ISO1 = 2
 local V2GTP_EXIDISSECTOR_ISO2 = 3
 
-local v2gtp_exidissector_tab =  {
-    { 1, "Default", V2GTP_EXIDISSECTOR_DEFAULT },
-    { 2, "DIN", V2GTP_EXIDISSECTOR_DIN },
-    { 3, "ISO1", V2GTP_EXIDISSECTOR_ISO1 },
-    { 4, "ISO2", V2GTP_EXIDISSECTOR_ISO2 },
+local v2gtp_exidissector_tab = {
+	{ 1, "Default", V2GTP_EXIDISSECTOR_DEFAULT },
+	{ 2, "DIN", V2GTP_EXIDISSECTOR_DIN },
+	{ 3, "ISO1", V2GTP_EXIDISSECTOR_ISO1 },
+	{ 4, "ISO2", V2GTP_EXIDISSECTOR_ISO2 },
 }
 
 v2gtp_protocol.prefs.exidissector = Pref.enum(
-    "EXI Dissector", -- label
-    V2GTP_EXIDISSECTOR_DEFAULT, -- default value
-    "Dissector to use for EXI", -- description
-    v2gtp_exidissector_tab, -- enum table
-    false -- show as combo box
+	"EXI Dissector", -- label
+	V2GTP_EXIDISSECTOR_DEFAULT, -- default value
+	"Dissector to use for EXI", -- description
+	v2gtp_exidissector_tab, -- enum table
+	false -- show as combo box
 )
 
 v2gtp_protocol.fields = SDP
 
 function set_contains(set, key)
-    return set[key] ~= nil
+	return set[key] ~= nil
 end
 
 function get_sdp_version(value)
-    local version = "RESERVED"
-    if value == 0x01 or bit32.bxor(value, 0xFF) == 0x01 then version = "V2GTP Version 1" end
+	local version = "RESERVED"
+	if value == 0x01 or bit32.bxor(value, 0xFF) == 0x01 then
+		version = "V2GTP Version 1"
+	end
 
-    return version
+	return version
 end
 
 function get_sdp_payload_type(type)
-    local type_name = "UNKNOWN"
-
-    local types = {
-    [0x9000] = "SDP REQUEST",
-    [0x9001] = "SDP RESPONSE",
-    [0x9004] = "SDP EMSP REQUEST",
-    [0x9005] = "SDP EMSP RESPONSE",
-    [0x8001] = "EXI ENCODED",
-    [0x9002] = "RESERVED",
-    [0X9003] = "RESERVED",
-    }
-
-    if set_contains(types, type) then return types[type] end
-
-    if type < 0x8000 then type_name = "RESERVED"
-    elseif type >= 0x8001 and type <= 0x8FFF then type_name = "RESERVED"
-    elseif type >= 0x9006 and type <= 0x9FFF then type_name = "RESERVED"
-    elseif type >= 0xA000 and type <= 0xFFFF then type_name = "MFG SPECIFIC"
-    end
-
-    return type_name
+	local type_name = "UNKNOWN"
+
+	local types = {
+		[0x8001] = "EXI ENCODED",
+		[0x8002] = "Part20Mainstream",
+		[0x8003] = "Part20ACMainstream",
+		[0x8004] = "Part20DCMainstream",
+		[0x8005] = "Part20ACDPMainstream",
+		[0x8006] = "Part20WPTMainstream",
+		[0x8101] = "ScheduleRenegotiation",
+		[0x8102] = "MeteringConfirmation",
+		[0x8103] = "ACDPSystemStatus",
+		[0x8104] = "ParkingStatus",
+		[0x9000] = "SDP REQUEST",
+		[0x9001] = "SDP RESPONSE",
+		[0x9002] = "SDP REQUEST WIRELESS",
+		[0X9003] = "SDP RESPONSE WIRELESS",
+		[0x9004] = "SDP EMSP REQUEST",
+		[0x9005] = "SDP EMSP RESPONSE",
+	}
+
+	if set_contains(types, type) then
+		return types[type]
+	end
+
+	if type < 0x8000 then
+		type_name = "RESERVED"
+	elseif type >= 0x8001 and type <= 0x8FFF then
```

Comment:
By addition of the ISO -20 codes, this now starts from 0x8007, but you return early anyway.
```suggestion
	elseif type >= 0x8007 and type <= 0x8FFF then
```